### ISSUE TITLE
Carousel: get current slide

### ISSUE THREAD

**Issue by lukaszprus** [ng-bootstrap/ng-bootstrap#3094]
https://ng-bootstrap.github.io/#/components/carousel/api

There's no method to get the current slide in the API. Is there any reason for that?

How do I get the current slide Id? The slide event isn't the best way to do it.

**Comment by pkozlowski-opensource** [comment]
How about `activeId`? 
https://stackblitz.com/edit/angular-xcv6dg?file=app/carousel-basic.html

**Comment by lukaszprus** [comment]
Thanks Paweł!

Correct me if I'm wrong but it looks like it's not the official API way of doing it? API states that activeId is just an input property so it gives no guarantee that the carousel component's activeId property always holds the current slide. I'm a bit worried about future implementation changes. For example, if one day you decide to go for `@Input('activeId') activeIdRenamed: string;`instead of `@Input() activeId: string;` in https://github.com/ng-bootstrap/ng-bootstrap/blob/master/src/carousel/carousel.ts

**Comment by pkozlowski-opensource** [comment]
@lukaszprus you are correct that it is a kind of "grey area" here... I'm marking it as a feature request so a proper method can be added to the API (ex. `getActiveId()`). PRs welcomed.

**Comment by lukaszprus** [comment]
You might want to consider making it a two-way binding: `[(activeId)]`

**Comment by pra-ver** [comment]
the issue with using activeID is the ID sets incorrect if used inside a tab.
For example, i add the carousel in a tab/route. slide by clicking next. ID is shown ngb-slide-0, ngb-slide-1, ngb-slide-2   
Now when i switch tabs and come back to the carousel, the id are now shown as ngb-slide-3,ngb-slide-4,ngb-slide-5

Heres a stackblitz
https://stackblitz.com/edit/angular-xcv6dg-1b9ws2

**Comment by pra-ver** [comment]
this works fine in an older version but has an issue in the latest version.
heres the fiddle with old version
https://angular-as4xyq.stackblitz.io

**Comment by hamleet** [comment]
Solution
https://github.com/ng-bootstrap/ng-bootstrap/issues/1397#issuecomment-619828208
